data/CMakeLists.txt

```diff
@@ -87,53 +87,35 @@ IF(WIN32)
-#-----------------------------------------------------------------------------
-IF(NOT VISIT_DATA_ARCHIVER_NAME)
-    list(APPEND VISIT_DATA_ARCHIVER_NAME "7z" "7za" "7zr" "p7zip")
-ENDIF()
+FIND_PACKAGE(Python3)
 
-#-----------------------------------------------------------------------------
-# Find the archiver exectuable
-#-----------------------------------------------------------------------------
-IF(WIN32)
-    SET(PF1 "PROGRAMFILES")
-    SET(PF2 "PROGRAMFILES(X86)")
-    FIND_PROGRAM(ARCHIVER_EXE NAMES ${VISIT_DATA_ARCHIVER_NAME}
-        PATHS ${VISIT_SEVEN_ZIP_DIR} 
-              $ENV{${PF1}}/7-Zip 
-              $ENV{${PF2}}/7-Zip)
-    UNSET(PF2)
-    UNSET(PF1)
-ELSE(WIN32)
-    FIND_PROGRAM(ARCHIVER_EXE NAMES ${VISIT_DATA_ARCHIVER_NAME}
-                              PATHS ${VISIT_SEVEN_ZIP_DIR}
-                              PATH_SUFFIXES bin)
-ENDIF(WIN32)
+#EXECUTE_PROCESS(COMMAND "${PYTHON3_EXECUTABLE}" "-c" "import tarfile, lzma"
+#    COMMAND_ERROR_IS_FATAL ANY)
+EXECUTE_PROCESS(COMMAND "${Python3_EXECUTABLE}" "-c" "import tarfile, lzma"
+    RESULT_VARIABLE PYTHON_TARFILE_AND_LZMA_MODULES_NOT_FOUND
+    ERROR_QUIET OUTPUT_QUIET)
+IF(PYTHON_TARFILE_AND_LZMA_MODULES_NOT_FOUND)
+    MESSAGE(FATAL_ERROR "python3 with tarfile and lzma modules is required")
+ENDIF()
 
 #-----------------------------------------------------------------------------
-# Set arguments for archiver creation, extraction and list convenience targets
-#-----------------------------------------------------------------------------
-IF("${ARCHIVER_EXE}" STREQUAL "ARCHIVER_EXE-NOTFOUND")
-    MESSAGE(WARNING "Archiver \"${VISIT_DATA_ARCHIVER_NAME}\" not found, data files cannot be extracted. Try setting VISIT_SEVEN_ZIP_DIR to location of 7-zip package, and VISIT_DATA_ARCHIVER_NAME to correct 7-zip executable name (eg, 7za, p7zip, 7z).")
-    #--------------------------------------------------------------------------
-    # CMake target to handle make-time requests to build targets that require
-    # 7z archiver yet none was provided. The test "" forces make-time failure.
-    #--------------------------------------------------------------------------
-    ADD_CUSTOM_TARGET(missing-7z-archiver ALL
-        COMMAND ${CMAKE_COMMAND} -E echo ""
-        COMMAND ${CMAKE_COMMAND} -E echo "No 7z archiver"
-        COMMAND ${CMAKE_COMMAND} -E echo ""
-        COMMAND test "")
-ELSE("${ARCHIVER_EXE}" STREQUAL "ARCHIVER_EXE-NOTFOUND")
-    SET(ARCHIVER_CMD ${ARCHIVER_EXE})
-    SET(ARCHIVER_XARGS x -y)                 # Expand archive args
-    SET(ARCHIVER_CARGS a -y -m0=lzma2 -mx=9) # Create archive args
-    SET(ARCHIVER_LARGS l -y)                 # List archive args
-ENDIF("${ARCHIVER_EXE}" STREQUAL "ARCHIVER_EXE-NOTFOUND")
+# Set up path to archiver executable
+#-----------------------------------------------------------------------------
+IF(NOT VISIT_DATA_ARCHIVER_CMD)
+    SET(ARCHIVER_CMD ${Python3_EXECUTABLE} ./visit_pytar.py)
```

Comment:
Should use '${CMAKE_CURRENT_SOURCE_DIR}/visit_pytar.py'

Reply:
Is it ok to keep separate from rest of VisIt? I think its small enough and independent enough that it can be.

Reply:
CMAKE_CURRENT_SOURCE_DIR has nothing to do with VisIt.  It simply refers to the current directory being processed by CMake, in this case the directory containing all the `tar.xz` files. 

If you want to build the data directory out-of-source, you need to have the correct location of `visit_pytar.py`  which isn't `./` for out-of-source builds. 

I prefer not to pollute my repo directories with build artefacts.